lib/iris/fileformats/pp.py

```diff
@@ -1037,9 +1037,11 @@ def _data_bytes_to_shaped_array(data_bytes, lbpack, boundary_packing,
         # Reform in row-column order
         data.shape = data_shape
 
-    # Mask the array?
+    # Convert mdi to NaN.
     if mdi in data:
-        data = array_masked_to_nans(data, data == mdi)
```

Comment:
If we pay attention to the name of `array_masked_to_nans`, it tells us that it will convert a masked array to a nan-filled array. If we don't have masked data (such as here), let's just not use it and instead do the processing we actually need to. That is, we cast int -> float if necessary and replace `mdi` values with `NaN`.